pkg/virt-handler/vm.go

```diff
@@ -186,11 +186,79 @@ type virtLauncherCriticalNetworkError struct {
 
 func (e *virtLauncherCriticalNetworkError) Error() string { return e.msg }
 
-func (d *VirtualMachineController) startDomainNotifyPipe(vmi *v1.VirtualMachineInstance) (net.Listener, error) {
+func handleDomainNotifyPipe(domainPipeStopChan chan struct{}, ln net.Listener, virtShareDir string, vmi *v1.VirtualMachineInstance) {
+
+	fdChan := make(chan net.Conn, 100)
+
+	// Listen for new connections,
+	// Close listener and exit when stop encountered
+	go func(vmi *v1.VirtualMachineInstance, ln net.Listener, domainPipeStopChan chan struct{}) {
```

Comment:
I think only one goroutine will stop if we use one stop `chan` for two goroutines. Isn't it better to use context?

Reply:
closing a channel should signal every goroutine that attempts to read from the channel. this is how our informers work for example.

Reply:
Here's a working example to illustrate the point. 

```
package main
 
import (
        "fmt"
        "sync"
        "time"
)
 
func main() {
        stopChan := make(chan struct{})
        wg := sync.WaitGroup{}
 
        wg.Add(2)
        go func(stop chan struct{}, wg *sync.WaitGroup) {
                for {
                        select {
                        case <-stop:
                                fmt.Println("done1")
                                wg.Done()
                                return
                        default:
                                time.Sleep(1)
                        }
                }
        }(stopChan, &wg)
 
        go func(stop chan struct{}, wg *sync.WaitGroup) {
                for {
                        select {
                        case <-stop:
                                fmt.Println("done2")
                                wg.Done()
                                return
                        default:
                                time.Sleep(1)
                        }
                }
        }(stopChan, &wg)
 
        close(stopChan)
       
        wg.Wait()
}     
```

Reply:
Thanks, should scroll little bit lower to see we close it.